pkg/controller/shard_handler.go

```diff
@@ -0,0 +1,79 @@
+package controller
+
+import (
+	"context"
+	typeV1 "github.com/istio-ecosystem/admiral-api/pkg/apis/admiral/v1"
+	"github.com/istio-ecosystem/admiral-sharding-manager/pkg/model"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ShardInteface interface {
+	CreateOrUpdateShard(ctx context.Context, shard *typeV1.Shard) (*typeV1.Shard, error)
+	DeleteShard(ctx context.Context, shard *typeV1.Shard) error
+}
+
+type ShardHandler struct {
+	Config         *model.ShardingManagerConfig
+	ShardNamespace string
+}
+
+func NewShardHandler(ShardConfig *model.ShardingManagerConfig, namespace string) ShardHandler {
+	shardHandler := ShardHandler{
+		Config:         ShardConfig,
+		ShardNamespace: namespace,
+	}
+
+	return shardHandler
+}
+
+func (sh ShardHandler) CreateOrUpdateShard(ctx context.Context, shard *typeV1.Shard) (*typeV1.Shard, error) {
```

Comment:
Can we make this a pointer receiver, so that if need be we can modify the config if required?

Reply:
can we have a different method for create and update?